From 529b4596617caa47f3aeb7e058a5e2a8b3f3aee6 Mon Sep 17 00:00:00 2001
From: Yong Chen <yong.a.chen@intel.com>
Date: Tue, 22 Oct 2013 15:29:20 +0800
Subject: Fix RenderScriptGL setSurfaceTexture problem

BZ: 129660

RS jni does not implement rsnContextSetSurfaceTexture(),
so use nContextSetSurface() for setSurfaceTexture().

Category: aosp improvement
Domain: AOSP-RenderScript
Origin: internal
Upstream-Candidate: yes

Change-Id: I32503944638244244a309788b5efa6ae685956ce
Depends-Change-Id: If9b09ccaca17da7eef0240676a11376c54f41799
Orig-MCG-Change-Id: I87085709016391391824ddd5537f3b8d7276dc16
Signed-off-by: Yong Chen <yong.a.chen@intel.com>
---
 .../java/android/renderscript/RenderScriptGL.java  |    6 +++++-
 1 files changed, 5 insertions(+), 1 deletions(-)

diff --git a/graphics/java/android/renderscript/RenderScriptGL.java b/graphics/java/android/renderscript/RenderScriptGL.java
index bac9c68..8549edb 100644
--- a/graphics/java/android/renderscript/RenderScriptGL.java
+++ b/graphics/java/android/renderscript/RenderScriptGL.java
@@ -232,9 +232,13 @@ public class RenderScriptGL extends RenderScript {
         validate();
         //android.util.Log.v("rs", "set surface " + sur + " w=" + w + ", h=" + h);
 
+        Surface s = null;
+        if (sur != null) {
+            s = new Surface(sur);
+        }
         mWidth = w;
         mHeight = h;
-        nContextSetSurfaceTexture(w, h, sur);
+        nContextSetSurface(w, h, s);
     }
 
     /**
-- 
1.7.4.1

